ext/drreg/drreg.c

```diff
@@ -65,17 +65,65 @@
 
 #define PRE instrlist_meta_preinsert
 
-/* This should be pretty hard to exceed as there aren't this many GPRs */
-#define MAX_SPILLS (SPILL_SLOT_MAX + 8)
+/* drreg uses DR spill slots by default if no addressable TLS
+ * storage is available. Since these slots are shared also with clients,
+ * it could be the case that the client uses a lot of these slots, resulting
+ * in an insufficient number of available slots for drreg to function properly.
+ *
+ * In order to address this concern, we use a hard-coded max on the number
```

Comment:
This is misleading IMO. The tool should reserve enough num_spill_slots, as pointed out in the docs. The MAX_SPILLS number here is really just a hardcoded limit on how many slots can be tracked. I admit the number has confused me as well, in particular it suggests that it needs to be tied to SPILL_SLOT_MAX, which is actually not a hard requirement. I would really just delete the entire second paragraph except the "pretty hard to exceed..."

Reply:
Resolving this because it relates to the next point.